peps/pep-9999.rst

```diff
@@ -0,0 +1,546 @@
+PEP: 9999
+Title: Classes & protocols: Read-only attributes
+Author: <REQUIRED: list of authors' real names and optionally, email addrs>
+Sponsor: Carl Meyer <carl@oddbird.net>
+Discussions-To: <REQUIRED: URL of current canonical discussion thread>
+Status: Draft
+Type: Standards Track
+Topic: Typing
+Created: ?-Nov-2024
+Python-Version: 3.?
+
+
+Abstract
+========
+
+:pep:`705` introduced the :external+py3.13:data:`typing.ReadOnly` type qualifier
+to allow defining read-only :class:`typing.TypedDict` items.
+
+This PEP proposes expanding the scope of ``ReadOnly`` to class and protocol
+attributes, as a single concise way to mark them read-only. Some parity changes
+are also made to :data:`typing.Final`.
+
+Akin to :pep:`705`, it makes no Python grammar changes. Correct usage of
+read-only attributes is intended to be enforced only by static type checkers.
+
+
+Motivation
+==========
+
+Python type system lacks a single concise way to mark an :term:`attribute` read-only.
+This feature is common in other object-oriented languages (such as `C# <https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/readonly>`_),
```

Comment:
To establish that it is "common", would be useful to mention one or two other languages. In particular, TypeScript (which has `readonly`) would be nice to mention as it also has a gradual type system.

Reply:
Perhaps wording it as
```suggestion
This feature is present in other object-oriented languages (such as `C# <https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/readonly>`_),
```
would be better; claiming it is common is somewhat bold

Reply:
I think that's fine, but either way mentioning TypeScript would be useful!